File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/main_program/fragmentPutOut/PutOutHttp.java
package com.example.nightwingky.monstone.main_program.fragmentPutOut;

import com.example.nightwingky.monstone.myConst.URLConst;

import java.io.IOException;

import okhttp3.FormBody;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by nightwingky on 17-3-19.
 */

public class PutOutHttp {

    public static String putOut(String name, String price, String catalog) throws IOException {
        OkHttpClient okHttpClient = new OkHttpClient();

        FormBody formBody = new FormBody.Builder()
                .add("product", name)
                .add("seller", URLConst.getUser_id())
                .add("price", price)
                .add("catalog", catalog)
                .build();

        Request request = new Request.Builder()
                .url(URLConst.getUser_put_out_url())
                .post(formBody)
                .build();

        Response response = okHttpClient.newCall(request).execute();

        if (response.isSuccessful()) {
            return response.body().string();
        } else {
            return null;
        }
    }
}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/main_program/fragmentPutOut/PutOutHttp.kt
package com.graduation.nightwingky.book_drifting.monstone.main_program.fragmentPutOut

import com.graduation.nightwingky.book_drifting.monstone.myConst.URLConst

import java.io.IOException

import okhttp3.FormBody
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response

/**
 * Created by nightwingky on 17-3-19.
 */

object PutOutHttp {

    @Throws(IOException::class)
    fun putOut(name: String, price: String, catalog: String): String? {
        val okHttpClient = OkHttpClient()

        val formBody = FormBody.Builder()
                .add("product", name)
                .add("seller", URLConst.user_id)
                .add("price", price)
                .add("catalog", catalog)
                .build()

        val request = Request.Builder()
                .url(URLConst.user_put_out_url)
                .post(formBody)
                .build()

        val response = okHttpClient.newCall(request).execute()

        return if (response.isSuccessful) {
            response.body()!!.string()
        } else {
            null
        }
    }
}
